static/app/views/performance/newTraceDetails/traceSearch/traceSearchEvaluator.tsx

```diff
@@ -74,10 +74,19 @@ export function searchInTraceTreeTokens(
 
   let i = 0;
   let matchCount = 0;
-  const count = tree.list.length;
   const resultsForSingleToken: TraceSearchResult[] = [];
 
   function searchSingleToken() {
+    // TODO Abdullah Khan: This implementation can be optimized;
```

Comment:
@Abdkhan14 It sounds to me like we should reverse this and just have the search logic traverse through the entire tree. The nice benefit of doing it like that is that we can preserve the tree shape for sections where we don't have matches as opposed to expanding everything.

In pseudo code, I think all that it would take is to iterate over the tree, mark matches and then sort them by the order in the tree. If you want, I can help you make the change

Reply:
Hey @JonasBa,

Thank you for the optimized implementation here: https://github.com/getsentry/sentry/pull/97276

I'll be merging your version on top of the improved tests, in a followup PR